google-cloud-storage/src/main/java/com/google/cloud/storage/spi/v1/HttpStorageRpc.java

```diff
@@ -562,6 +565,38 @@ public StorageObject get(StorageObject object, Map<Option, ?> options) {
     }
   }
 
+  @Override
+  public StorageObject restore(StorageObject object, Map<Option, ?> options) {
+    Span span = startSpan(HttpStorageRpcSpans.SPAN_NAME_RESTORE_OBJECT);
+    Scope scope = tracer.withSpan(span);
+    try {
+      Storage.Objects.Restore restore =
+          storage
+              .objects()
+              .restore(object.getBucket(), object.getName(), object.getGeneration(), object);
+      return restore
+          .setProjection(DEFAULT_PROJECTION)
+          .setIfMetagenerationMatch(Option.IF_METAGENERATION_MATCH.getLong(options))
+          .setIfMetagenerationNotMatch(Option.IF_METAGENERATION_NOT_MATCH.getLong(options))
+          .setIfGenerationMatch(Option.IF_GENERATION_MATCH.getLong(options))
+          .setIfGenerationNotMatch(Option.IF_GENERATION_NOT_MATCH.getLong(options))
+          .setCopySourceAcl(Option.COPY_SOURCE_ACL.getBoolean(options))
```

Comment:
This is an interesting addition to the API; surprised it needs to be added if you're restoring an object.